### PR TITLE
Use HorizontalGrid and VerticalGrid string values to specify GridItemsLayout in XAML

### LINKED ISSUE #5577: CollectionView XAML API suggestion
### Description

I love `CollectionView`. Particularly the ability to specify different layouts. However, I'd like to see it be easier to specify layouts from XAML. This isn't a bug, more a suggestion.

By default `CollectionView` produces a vertical list layout:

```
<CollectionView ItemsSource="{Binding Monkeys}" />
```

If we want to change that to a horizontal list, we have to do the following:

```
<CollectionView ItemsSource="{Binding Monkeys}"
                            ItemsLayout="{x:Static ListItemsLayout.HorizontalList}">
```

Personally, I think specifying the `ItemsLayout` above like that is a bit ugly.

A vertical grid is:

```
        <CollectionView ItemsSource="{Binding Monkeys}">
            <CollectionView.ItemsLayout>
               <GridItemsLayout Orientation="Vertical"
                                Span="2" />
            </CollectionView.ItemsLayout>
```

A horizontal grid is:

```
        <CollectionView ItemsSource="{Binding Monkeys}">
            <CollectionView.ItemsLayout>
               <GridItemsLayout Orientation="Horizontal"
                                Span="4" />
```

My suggestions/questions are:

1. `ListItemsLayout` defines static `VerticalList` and `HorizontalList` members that can be consumed from XAML. While I think this leads to ugly XAML, should `GridItemsLayout` define some static members for common grid layouts?
1. Is a better solution to implement a type converter so users can do something like `ItemsLayout="HorizontalList"` or `ItemsLayout="VerticalList"`? Personally, I like this idea.
1. Extending above, should users be able to do something like `ItemsLayout="HorizontalGrid, 2"` or `ItemsLayout="VerticalGrid, 3"`? Not convinced about this, but thought I'd raise it anyway.

### PR DESCRIPTION
**Pull request by duracellko**
### Description of Change ###

PR extends ItemsLayoutTypeConverter to support HorizontalGrid and VerticalGrid with Span specified. Unit tests for ItemsLayoutTypeConverter are added.

With this change it is possible to specify following string values for ItemsLayout property in XAML:
- `HorizontalList` - ItemsLayout is set to `LinearItemsLayout.Horizontal`
- `VerticalList` - ItemsLayout is set to `LinearItemsLayout.Vertical`
- `HorizontalGrid` - ItemsLayout is set to `GridItemsLayout` with Horizontal Orientation.
- `VerticalGrid` - ItemsLayout is set to `GridItemsLayout` with VerticalOrientation.
- `HorizontalGrid, {span}` - ItemsLayout is set to `GridItemsLayout` with Horizontal Orientation and Span set to value of {span}.
- `VerticalGrid, {span}` - ItemsLayout is set to `GridItemsLayout` with Vertical Orientation and Span set to value of {span}.

### Issues Resolved ### 

- fixes #5577

### API Changes ###
None

### Platforms Affected ### 
All

### Behavioral/Visual Changes ###

None. Changes affect XAML only.

### Testing Procedure ###
Test-case 5577 in Controls Gallery.

### PR Checklist ###

- [ ] Targets the correct branch
- [ ] Tests are passing (or failures are unrelated)

### CODE REVIEW

File: Xamarin.Forms.Core/Items/ItemsLayoutTypeConverter.cs [hunk 1]
```diff
@@ -1,23 +1,51 @@
 ﻿using System;
+using System.Globalization;
 
 namespace Xamarin.Forms
 {
 	[Xaml.TypeConversion(typeof(IItemsLayout))]
 	public class ItemsLayoutTypeConverter : TypeConverter
 	{
+		private const string HorizontalGridPrefix = "HorizontalGrid,";
+		private const string VerticalGridPrefix = "VerticalGrid,";
```
Comment (StephaneDelcroix) [inline Xamarin.Forms.Core/Items/ItemsLayoutTypeConverter.cs]: don't include the comma in the prefix. you can even drop this const, as you use the value by string later on

**Comment by StephaneDelcroix** [comment]
please provide a better title and description, documenting the new format

File: Xamarin.Forms.Core/Items/ItemsLayoutTypeConverter.cs (hunk 1 above)
Comment (duracellko) [reply Xamarin.Forms.Core/Items/ItemsLayoutTypeConverter.cs]: I re-implemented the parsing. At first identifier is identified and then span is parsed, if string is longer.

File: Xamarin.Forms.Core/Items/ItemsLayoutTypeConverter.cs [hunk 2]
```diff
@@ -7,15 +8,46 @@ public class ItemsLayoutTypeConverter : TypeConverter
 	{
 		public override object ConvertFromInvariantString(string value)
 		{
-			if (value == "HorizontalList")
+			if (value == null)
```
Comment (jfversluis) [inline Xamarin.Forms.Core/Items/ItemsLayoutTypeConverter.cs]: null or should we check string.IsNullOrWhiteSpace? or string.IsNullOrEmpty?

Comment (duracellko) [reply Xamarin.Forms.Core/Items/ItemsLayoutTypeConverter.cs]: On empty string InvalidOperationException with text "Cannot convert "" into Xamarin.Forms.IItemsLayout" is thrown. I think it is more descriptive than "Value cannot be null."

Comment (jfversluis) [reply Xamarin.Forms.Core/Items/ItemsLayoutTypeConverter.cs]: I'm not sure if we mean the same thing. I was suggesting, shouldn't we change it to: `if (string.IsNullOrWhitespace(value))`

**Comment by jfversluis** [comment]
Hi @duracellko sorry for the long wait. Would you be able to resolve conflicts/rebase?

**Comment by duracellko** [comment]
> Hi @duracellko sorry for the long wait. Would you be able to resolve conflicts/rebase?

I will have a look at it on weekend.

**Comment by duracellko** [comment]
Rebased

**Comment by jfversluis** [comment]
Rebase and all tests look good, thanks! :)

**Comment by StephaneDelcroix** [comment]
adding DoNOTMERGE so @hartez can review

**Comment by hartez** [comment]
Awesome stuff, @duracellko!

**Pull request merged**